code/parachain/frame/staking-rewards/src/test/mod.rs

```diff
@@ -625,73 +625,267 @@ fn stake_in_case_of_not_zero_inflation_should_work() {
+			process_and_progress_blocks::<StakingRewards, Test>(1);
 
-		let rewards_pool = StakingRewards::pools(pool_id).expect("rewards_pool expected");
+			mint_assets([ALICE], [USDT::ID], USDT::units(100_000));
+			add_to_rewards_pot_and_assert(ALICE, STAKED_ASSET::ID, USDT::ID, USDT::units(100_000));
 
-		let reductions = rewards_pool
-			.rewards
-			.iter()
-			.map(|(asset_id, _reward)| (*asset_id, 0))
-			.try_collect()
-			.expect("reductions expected");
+			let staked_amount = STAKED_ASSET::units(5);
+			let extended_amount = STAKED_ASSET::units(6);
+			let existential_deposit = 1_000_u128;
 
-		let stake = StakingRewards::stakes(1, 0);
-		assert_eq!(
-			stake,
-			Some(Stake {
-				fnft_instance_id: 0,
-				reward_pool_id: pool_id,
-				stake: amount + extend_amount,
-				share: boosted_amount + extend_amount,
-				reductions,
-				lock: Lock {
-					started_at: 12,
-					duration: duration_preset,
-					unlock_penalty: rewards_pool.lock.unlock_penalty,
+			mint_assets(
+				[BOB],
+				[STAKED_ASSET::ID],
+				staked_amount + extended_amount + existential_deposit,
+			);
+
+			let fnft_instance_id = stake_and_assert::<Test, runtime::Event>(
+				BOB,
+				STAKED_ASSET::ID,
+				staked_amount,
+				ONE_MINUTE,
+			);
+
+			process_and_progress_blocks::<StakingRewards, Test>(10);
+
+			assert_extrinsic_event::<Test, runtime::Event, _, _, _>(
+				StakingRewards::extend(
+					Origin::signed(BOB),
+					STAKING_FNFT_COLLECTION_ID,
+					fnft_instance_id,
+					extended_amount,
+				),
+				crate::Event::<Test>::StakeAmountExtended {
+					fnft_collection_id: STAKING_FNFT_COLLECTION_ID,
+					fnft_instance_id,
+					amount: extended_amount,
 				},
-			})
-		);
-		assert_eq!(balance(staked_asset_id, &staker), existential_deposit);
-		assert_eq!(
-			balance(staked_asset_id, &FinancialNft::asset_account(&1, &0)),
-			amount + extend_amount
-		);
-		assert_eq!(
-			balance(XPICA::ID, &FinancialNft::asset_account(&1, &0)),
-			boosted_amount + extend_amount
-		);
-		assert_last_event::<Test, _>(|e| {
-			matches!(e.event,
-            Event::StakingRewards(crate::Event::StakeAmountExtended { fnft_collection_id, fnft_instance_id, amount})
-            if fnft_collection_id == 1_u128 && fnft_instance_id == 0_u64 && amount == extend_amount)
+			);
+
+			let stake_after_extend =
+				Stakes::<Test>::get(STAKING_FNFT_COLLECTION_ID, fnft_instance_id)
+					.expect("stake should exist");
+
+			let rewards_pool =
+				StakingRewards::pools(STAKED_ASSET::ID).expect("rewards pool should exist");
+
+			assert_eq!(
+				stake_after_extend,
+				Stake {
+					fnft_instance_id,
+					reward_pool_id: STAKED_ASSET::ID,
+					stake: staked_amount + extended_amount,
+					share: Pallet::<Test>::boosted_amount(
+						rewards_pool.lock.duration_presets[&ONE_MINUTE],
+						staked_amount + extended_amount
+					)
+					.expect("boosted amount calculation should not fail"),
+					// TODO(benluelo): Figure out a way to not have a magic number here (i.e. how to
```

Comment:
Having a magic number is ok in a unit test as long as the calculation is explained in a comment. Calculating it just hard codes the test to the current implementation of the protocol

Reply:
Think of a property test like rsTest for example